File: my-utils/src/main/java/com/sleticalboy/util/SPUtils.java
package com.sleticalboy.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;

/**
 * Created by Android Studio.
 * Date: 12/27/17.
 *
 * @author sleticalboy
 */
public class SPUtils {

    private static SharedPreferences sSharedPreferences;

    private SPUtils() {
    }

    public static void init(@NonNull Context context) {
        if (sSharedPreferences == null) {
            sSharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        }
    }

    public static void putInt(String key, int value) {
        sSharedPreferences.edit().putInt(key, value).apply();
    }

    public static int getInt(String key, int defValue) {
        return sSharedPreferences.getInt(key, defValue);
    }

    public static void putBoolean(String key, boolean value) {
        sSharedPreferences.edit().putBoolean(key, value).apply();
    }

    public static boolean getBoolean(String key, boolean defValue) {
        return sSharedPreferences.getBoolean(key, defValue);
    }

    public static void putString(String key, String value) {
        sSharedPreferences.edit().putString(key, value).apply();
    }

    public static String getString(String key, String defValue) {
        return sSharedPreferences.getString(key, defValue);
    }
}

File: my-utils/src/main/java/com/sleticalboy/util/SPUtils.kt
package com.sleticalboy.util

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

/**
 * Created by Android Studio.
 * Date: 12/27/17.
 *
 * @author sleticalboy
 */
object SPUtils {

    private var sp: SharedPreferences? = null

    @JvmStatic
    fun init(context: Context) {
        if (sp == null) {
            sp = context.getSharedPreferences(context.packageName + "_preferences", Context.MODE_PRIVATE)
        }
    }

    @JvmStatic
    fun putInt(key: String, value: Int) {
        sp!!.edit().putInt(key, value).commit()
    }

    @JvmStatic
    fun getInt(key: String, defValue: Int): Int {
        return sp!!.getInt(key, defValue)
    }

    @JvmStatic
    fun putBoolean(key: String, value: Boolean) {
        sp!!.edit().putBoolean(key, value).commit()
    }

    @JvmStatic
    fun getBoolean(key: String, defValue: Boolean): Boolean {
        return sp!!.getBoolean(key, defValue)
    }

    @JvmStatic
    fun putString(key: String, value: String) {
        sp!!.edit().putString(key, value).commit()
    }

    @JvmStatic
    fun getString(key: String, defValue: String?): String? {
        return sp!!.getString(key, defValue)
    }
}
